### ISSUE TITLE
Query-time profile switch

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#187]
It would be nice to be able to decide on a routing setting with several profiles, and then offer a way to switch profile right in the json input, e.g. by providing a `profile` key to vehicles.

When using several `osrm-routed` profiles, this can already be done by adjusting the `-a` and `-p` flags, but this requires to now in advance what profile should be used. We could instead have a generic way to describe OSRM instances from the command-line perspective, something like:

```
vroom -a car:0.0.0.0 -p car:5000 -a bike:0.0.0.0 -p bike:5001 [...]
```

When using `libosrm`, we could rely on named shared memory datasets to be used as profile names, so this would be a follow-up from #181.

Notes:
- we should provide generic ways to setup the configuration directly from C++ for use with `libvroom`
- defining and using vehicles with several profiles in the same problem instance is out of scope for this ticket

**Comment by jcoupey** [comment]
Couple thoughts on this.

1. The `-p` flag could probably be removed altogether as the port can be part of the address
2. The `-l` flag to toogle using `osrm-routed` or `libosrm` is not that clear, especially in the light of #192. We should remove it and have a `-r` flag to explicitly describe the routing engine to pick among:
  - osrm (default), uses `osrm-routed`
  - libosrm
  - ... (extensible at will)
3. The (undocumented) `-m` flag is replaced by a profile description in the API at the vehicle level, probably `vehicle.profile` defaulting to `car`. Providing an input file with vehicles that have different profiles is (at least for now) an error case.

**Comment by jcoupey** [comment]
This landed with #196. The "Usage" wiki page should be updated after the next release to describe the new options usage.

Note: point 1. above is not possible as the port is directly used by `boost::asio` to handle the request. The updated syntax for `-a` and `-p` is the one described initially above.
